clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -96,6 +98,71 @@ public MockProducer(boolean autoComplete, Partitioner partitioner, Serializer<K>
     }
 
     /**
+     * Needs to be called before any of the other transaction methods. Assumes that
+     * the transactional.id is specified in the producer configuration.
+     *
+     * This method does the following:
+     *   1. Ensures any transactions initiated by previous instances of the producer
+     *      are completed. If the previous instance had failed with a transaction in
+     *      progress, it will be aborted. If the last transaction had begun completion,
+     *      but not yet finished, this method awaits its completion.
+     *   2. Gets the internal producer id and epoch, used in all future transactional
+     *      messages issued by the producer.
+     *
+     * @throws IllegalStateException if the TransactionalId for the producer is not set
+     *         in the configuration.
+     */
```

Comment:
I don't think we need the docs in the `MockProducer`. It's enough to have them in `KafkaProducer`.